### PR TITLE
Create url.d.ts

### PR DESCRIPTION
**Pull request by zunyan**
add d.ts to support typescript build system

### CODE REVIEW

**Comment by DG-King** [comment]
it is fail

File: url.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+export function parse(url: string, parseQueryString: string, slashesDenoteHost: string): Url;
+export function resolve(source: string, relative: string): string;
+export function resolveObject(source: string, relative: string): Url;
+export function format(obj: string | url): string;
```
Comment (ArthurClemens) [inline url.d.ts:6]: This should be:
```
export function format(urlObj: Url): string;
```

File: url.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+export function parse(url: string, parseQueryString: string, slashesDenoteHost: string): Url;
```
Comment (ArthurClemens) [inline url.d.ts:3]: This should be:
```
export function parse(url: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
```

**Comment by ljharb** [comment]
This isn't a TypeScript project. Types are best added in a DefinitelyTyped package, rather than conflating the semver of real JS code with the semver of types.

Please feel free to add me as an owner on the DT PR, and I'll review as needed.

**Pull request closed without merging**